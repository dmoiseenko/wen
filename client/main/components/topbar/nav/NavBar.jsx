import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import LogoutContainer from '../logout/Logout.container';
import NavLinkContainer from './NavLink.container';
import * as routes from '../../../../../common/routes';


export default function NavBar({ isOpen }) {
  const className = classnames('nav-right nav-menu', {
    'is-active': isOpen
  });

  return (
    <div className={className}>
      <NavLinkContainer
        route={routes.home}
        label="Home"
      />
      <NavLinkContainer
        route={routes.notes}
        label="Notes"
      />
      <NavLinkContainer
        route={routes.graph}
        label="Graph"
      />
      <NavLinkContainer
        route={routes.about}
        label="About"
      />
      <div className="nav-item">
        <LogoutContainer />
      </div>
    </div>
  );
}

NavBar.propTypes = {
  isOpen: PropTypes.bool.isRequired
};
