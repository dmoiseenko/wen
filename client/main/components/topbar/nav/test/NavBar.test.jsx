import React from 'react';
import { shallow } from 'enzyme';

import NavBar from '../NavBar';


it('should render closed if isOpen false', () => {
  const props = {
    isOpen: false
  };

  const wrapper = shallow(
    <NavBar {...props} />
  );

  expect(wrapper.node).toMatchSnapshot();
});

it('should render opened if isOpen true', () => {
  const props = {
    isOpen: true
  };

  const wrapper = shallow(
    <NavBar {...props} />
  );

  expect(wrapper.node).toMatchSnapshot();
});
