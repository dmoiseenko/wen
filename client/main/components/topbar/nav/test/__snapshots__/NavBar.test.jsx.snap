// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render closed if isOpen false 1`] = `
<div
  className="nav-right nav-menu"
>
  <withRouter(Connect(NavigationLink))
    label="Home"
    route="/"
  />
  <withRouter(Connect(NavigationLink))
    label="Notes"
    route="/notes"
  />
  <withRouter(Connect(NavigationLink))
    label="Graph"
    route="/graph"
  />
  <withRouter(Connect(NavigationLink))
    label="About"
    route="/about"
  />
  <div
    className="nav-item"
  >
    <Connect(Logout) />
  </div>
</div>
`;

exports[`should render opened if isOpen true 1`] = `
<div
  className="nav-right nav-menu is-active"
>
  <withRouter(Connect(NavigationLink))
    label="Home"
    route="/"
  />
  <withRouter(Connect(NavigationLink))
    label="Notes"
    route="/notes"
  />
  <withRouter(Connect(NavigationLink))
    label="Graph"
    route="/graph"
  />
  <withRouter(Connect(NavigationLink))
    label="About"
    route="/about"
  />
  <div
    className="nav-item"
  >
    <Connect(Logout) />
  </div>
</div>
`;
